src/hardware/gpu.cpp

```diff
@@ -0,0 +1,23 @@
+#ifdef NMC_HAVE_GPU
+    #include <cuda_runtime.h>
+#endif
+
+namespace nest {
+namespace mc {
+namespace hw {
+
+#ifdef NMC_HAVE_GPU
+unsigned num_available_gpus() {
```

Comment:
Similar to above with naming; `num_available_gpus()` is kinda verbose; we're already under `hw::` and having 'available' in the name doesn't really add anything.